github/misc_test.go

```diff
@@ -168,3 +168,39 @@ func TestRepositoriesService_ListServiceHooks(t *testing.T) {
 		t.Errorf("Repositories.ListServiceHooks returned %+v, want %+v", hooks, want)
 	}
 }
+
+func TestParseURL(t *testing.T) {
+	const (
+		owner, repo = "bazelbuild", "bazel"
+	)
+	tests := []struct {
+		url                 string
+		wantOwner, wantRepo string
+		wantErr             bool
+	}{
+		{url: "https://github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "http://github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "https://www.github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/releases/tag/0.2.1", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/commit/19b...edc", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/archive/0.2.1.zip", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/", wantOwner: owner},
+		{url: "https://github.com/bazelbuild", wantOwner: owner},
+		{url: "https://mygithub.com/bazelbuild", wantErr: true},
+		{url: "https://github.com/", wantErr: true},
+		{url: "https://github.com", wantErr: true},
+	}
+
+	for _, test := range tests {
+		gotOwner, gotRepo, err := ParseURL(test.url)
+		if err != nil && !test.wantErr {
```

Comment:
The `wantErr` check here is ineffective. It verifies that `err` must be `nil` when `wantErr` is false. However, when `wantErr` is true, either `nil` or not `nil` value of `err` will pass without test failing.

As a result, it didn't catch that there's an inconsistency in the `{url: "https://github.com/", wantErr: true},` line. `ParseURL("https://github.com/")`, as is currently written, actually returns `nil` error.